contracts/Controller.sol

```diff
@@ -799,6 +882,62 @@ contract Controller is Initializable, OwnableUpgradeSafe, ReentrancyGuardUpgrade
         emit VaultSettled(_args.owner, _args.to, address(otoken), _args.vaultId, payout);
     }
 
+    /**
+     * @notice liquidate naked margin vault
+     * @dev can liquidate different vaults id in the same operate() call
+     * @param _args liquidation action arguments struct
+     */
+    function _liquidate(Actions.LiquidateArgs memory _args) internal notPartiallyPaused {
+        require(_checkVaultId(_args.owner, _args.vaultId), "Controller: invalid vault id");
+
+        // check if vault is undercollateralized
+        // the price is the amount of collateral asset to pay per 1 repaid debt(otoken)
+        // collateralDust is the amount of collateral that should be left in vault if not fully liquidated by liquidator
+        (MarginVault.Vault memory vault, bool isLiquidatable, uint256 price, uint256 collateralDust) = _isLiquidatable(
+            _args.owner,
+            _args.vaultId,
+            _args.roundId
+        );
+
+        // do not allow liquidating un-liquidatable vault
+        require(isLiquidatable, "Controller: can not liquidate vault");
```

Comment:
```suggestion
        require(isLiquidatable, "Controller: can not liquidate the vault");
```

Reply:
// do not allow liquidating un-liquidatable vault
this comment is redundant too